app/gui/view/graph-editor/src/component/node.rs

```diff
@@ -329,13 +329,19 @@ ensogl::define_endpoints_2! {
         /// Whether visualization was permanently enabled (e.g. by pressing the button).
         visualization_enabled    (bool),
         /// Visualization can be visible even when it is not enabled, e.g. when showing preview.
+        /// Visualization can be invisible even when enabled, e.g. when the node has an error.
         visualization_visible    (bool),
         visualization_path       (Option<visualization::Path>),
         expression_label_visible (bool),
-        bounding_box             (BoundingBox),
-        /// The [`display::object::Model::position`] of the Node. The position change events are
-        /// emitted after corresponding [`bounding_box`] change events.
+        /// The [`display::object::Model::position`] of the Node. Emitted when the Display Object
+        /// hierarchy is updated (see: [`ensogl_core::display::object::Instance::update`]).
         position                 (Vector2),
+        /// The bounding box of the Node. Contains the bounding box of the visualization if the
+        /// visualization is enabled and visible.
+        ///
+        /// Updated after any of [`position`], [`expression`], [`visualization_enabled`], or
+        /// [`visualization_visible`] is updated.
```

Comment:
From these cases only the first is non-trivial, so I would focus on it - So, let's put a note "please remember, that the position is not immediately updated, only during the Display Object hierarchy update".